clang/lib/Sema/SemaExpr.cpp

```diff
@@ -12120,8 +12120,7 @@ static void DiagnoseBadShiftValues(Sema& S, ExprResult &LHS, ExprResult &RHS,
     auto FXSema = S.Context.getFixedPointSemantics(LHSExprType);
     LeftSize = FXSema.getWidth() - (unsigned)FXSema.hasUnsignedPadding();
   }
-  llvm::APInt LeftBits(Right.getBitWidth(), LeftSize);
-  if (Right.uge(LeftBits)) {
+  if (Right.uge(LeftSize)) {
```

Comment:
I tried to understand why the old code used an APInt here.
What if you have something like ` x >> larger_than_64_bits_value`? I suspect that then the uge helper here will do a 64-bit unsigned compare. So it will truncate `Right` to 64 bits instead, right?

So I think the easiest way would be to compare to APInt:s. But make sure those are zero-extended to have a common size.
Maybe something like this:
```
  unsigned CompareBits = std::max(Right.getBitWidth(), 64);
  llvm::APInt LeftBits(CompareBits, LeftSize);
  if (Right.zext(CompareBits).uge(LeftBits)) {
```

Reply:
Oops, sorry. Now I think my previous comment was wrong. I missed the `getActiveBits()` check in APInt::ult. So it should be fine without complicating it that way.